### PR TITLE
Add profile picture to Book display and UI

### PR DESCRIPTION
**Pull request by abbymapes**
Add user profile picture from Google People API to book display 
- when viewing book likes
- when viewing friend's likes
- when viewing personal likes

Add bookshelf names to display for bookshelves

Update testing to match changes to display picture information 
- pass Friend objects to frontend instead of just names, etc.

Update AuthServlet and general display to output user's profile information from Google People API
- including their name and own profile picture (rather than generic picture)

Fix bug where user likes were stored by the email they signed in with (capitalization pattern included) and friend's contacts has user's email with different capitalization (wasn't recognizing as same email)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/utils/OAuthHelper.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  /**
+   * Creates a redirect URL for OAuth Servlets
+   *
+   * @return url String
+   */
+  public static String createRedirectUri() throws ServletException, IOException {
+    GenericUrl url =
+        new GenericUrl("https://8080-fabf4299-6bc0-403a-9371-600927588310.us-west1.cloudshell.dev");
```
Comment (marcusschwartz) [inline portfolio/src/main/java/com/google/sps/utils/OAuthHelper.java]: this url should probably be statically defined in one place.

File: portfolio/src/main/java/com/google/sps/utils/MemoryUtils.java [hunk 2]
```diff
@@ -377,18 +377,26 @@ public static void saveAggregateListData(
...
       String stemmedListName,
       List<String> items,
-      boolean newList) throws InvalidRequestException {
+      boolean newList)
+      throws InvalidRequestException {
     log.info("making storeInfo api request");
     RestTemplate restTemplate = new RestTemplate();
-    String urlString = "https://arliu-step-2020-3.wl.r.appspot.com/storeInfo?userID=" + userID + "&stemmedListName=" + stemmedListName + "&newList=" + newList;
+    String urlString =
+        "https://arliu-step-2020-3.wl.r.appspot.com/storeInfo?userID="
```
Comment (marcusschwartz) [inline portfolio/src/main/java/com/google/sps/utils/MemoryUtils.java:385]: this base url should be defined in one place

File: recommendations/src/main/java/com/google/sps/recommendations/Recommender.java [hunk 3]
```diff
@@ -181,9 +181,15 @@ private double increment(double e1, double e2, double error, double alpha) {
   }
 
   /**
+<<<<<<< HEAD:portfolio/src/main/java/com/google/sps/data/Recommender.java
```
Comment (marcusschwartz) [inline recommendations/src/main/java/com/google/sps/recommendations/Recommender.java:184]: merge mistake?

File: portfolio/src/main/java/com/google/sps/utils/OAuthHelper.java (hunk 1 above)
Comment (abbymapes) [reply portfolio/src/main/java/com/google/sps/utils/OAuthHelper.java]: @aliu917 I am going to statically define your URL in the class.

File: portfolio/src/main/java/com/google/sps/utils/MemoryUtils.java (hunk 2 above)
Comment (abbymapes) [reply portfolio/src/main/java/com/google/sps/utils/MemoryUtils.java:385]: @aliu917 I am going to statically define your URL in the class.

File: recommendations/src/main/java/com/google/sps/recommendations/Recommender.java (hunk 3 above)
Comment (abbymapes) [reply recommendations/src/main/java/com/google/sps/recommendations/Recommender.java:184]: @aliu917 Can I delete this?

**Pull request merged**